### PR TITLE
Add tests for uncovered code paths (tests only)

### PR DESCRIPTION
**Pull request by Schwarf**
**Description:**  
This PR adds new or extends existing unit tests to cover edge cases and close minor coverage gaps across various `networkit` submodules.  
No production code is modified.

### CODE REVIEW

File: networkit/cpp/clique/test/MaximalCliquesGTest.cpp [hunk 1]
```diff
@@ -12,6 +12,19 @@ namespace NetworKit {
 
 class MaximalCliquesGTest : public testing::Test {};
 
+TEST_F(MaximalCliquesGTest, testGetCliquesThrowsWithCallback) {
+    Graph graph(0, false, true, false);
+    MaximalCliques maxClique(graph, [&](const std::vector<node> &) {});
+    try {
+        maxClique.getCliques();
+        FAIL() << "Expected std::runtime_error";
```
Comment (fabratu) [inline networkit/cpp/clique/test/MaximalCliquesGTest.cpp:20]: You can use `EXPECT_THROW` here.

File: networkit/cpp/components/test/ConnectedComponentsGTest.cpp [hunk 2]
```diff
@@ -77,6 +77,18 @@ TEST_F(ConnectedComponentsGTest, testConnectedComponents) {
     EXPECT_EQ(1029u, cc.numberOfComponents());
 }
 
+TEST_F(ConnectedComponentsGTest, testParallelConnectedComponentsThowsForDirectedGraph) {
+    Graph G(2, false, true);
+    try {
+        ParallelConnectedComponents test(G);
+        FAIL() << "Expected std::runtime_error";
```
Comment (fabratu) [inline networkit/cpp/components/test/ConnectedComponentsGTest.cpp:84]: Same here.

**Review by fabratu — CHANGES_REQUESTED** [review]

File: networkit/cpp/clique/test/MaximalCliquesGTest.cpp (hunk 1 above)
Comment (Schwarf) [reply networkit/cpp/clique/test/MaximalCliquesGTest.cpp]: Thank you. Done

File: networkit/cpp/components/test/ConnectedComponentsGTest.cpp (hunk 2 above)
Comment (Schwarf) [reply networkit/cpp/components/test/ConnectedComponentsGTest.cpp]: Done

**Review by fabratu — APPROVED** [review]
